packages/gestalt/src/Checkbox.js

```diff
@@ -57,68 +64,104 @@ export default function Checkbox({
   let borderStyle = styles.border;
   if (!disabled && (checked || indeterminate)) {
     borderStyle = styles.borderDark;
-  } else if (hasError) {
+  } else if (hasError || errorMessage) {
     borderStyle = styles.borderError;
   }
 
   return (
-    <Box position="relative">
-      <input
-        checked={checked}
-        className={classnames(styles.input, {
-          [styles.inputEnabled]: !disabled,
-          [styles.inputSm]: size === 'sm',
-          [styles.inputMd]: size === 'md',
-        })}
-        disabled={disabled}
-        id={id}
-        name={name}
-        onBlur={() => setFocused(false)}
-        onChange={handleChange}
-        onClick={handleClick}
-        onFocus={() => setFocused(true)}
-        ref={inputElement}
-        type="checkbox"
-      />
-      <div
-        className={classnames(
-          borderStyle,
-          styles.check,
-          // eslint-disable-next-line no-nested-ternary
-          disabled
-            ? checked || indeterminate
-              ? colors.grayBg
-              : colors.lightGrayBg
-            : checked || indeterminate
-            ? colors.darkGrayBg
-            : colors.whiteBg,
-          {
-            [styles.checkEnabled]: !disabled,
-            [styles.checkFocused]: focused,
-            [styles.checkMd]: size === 'md',
-            [styles.checkSm]: size === 'sm',
-          }
-        )}
+    <Box>
+      <Box
+        alignItems="center"
+        display="flex"
+        justifyContent="start"
+        marginLeft={-1}
+        marginRight={-1}
       >
-        {(checked || indeterminate) && (
-          <Icon
-            accessibilityLabel=""
-            color="white"
-            icon={indeterminate ? 'dash' : 'check'}
-            size={size === 'sm' ? 8 : 12}
-          />
+        <Label htmlFor={id}>
+          <Box paddingX={1} position="relative">
+            <div
+              className={classnames(
+                borderStyle,
+                styles.check,
+                // eslint-disable-next-line no-nested-ternary
+                disabled
+                  ? colors.lightGrayBg
+                  : checked || indeterminate
+                  ? colors.darkGrayBg
+                  : colors.whiteBg,
+                {
+                  [styles.checkFocused]: focused,
+                  [styles.checkMd]: size === 'md',
+                  [styles.checkSm]: size === 'sm',
+                }
+              )}
+            >
+              {(checked || indeterminate) && (
+                <Icon
+                  accessibilityLabel=""
+                  color="white"
+                  icon={indeterminate ? 'dash' : 'check'}
+                  size={size === 'sm' ? 8 : 12}
+                />
+              )}
+            </div>
+            <input
```

Comment:
@AlbertCarreras this `input` should be before the `<div />`

Reply:
DONE